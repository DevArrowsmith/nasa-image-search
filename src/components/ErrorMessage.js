import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ErrorStyles = styled.h2`
margin: 20px 20px;
color: #D6D4D0;
`

const ErrorMessage = ({ errorParams }) => {
  return (
    <div data-testid="error-message">
      <ErrorStyles>
        {`${errorParams.errorStatus} Error: ${errorParams.errorText}`}
      </ErrorStyles>
    </div>
  )
};

ErrorMessage.propTypes = {
  errorParams: PropTypes.object
};

export default ErrorMessage;