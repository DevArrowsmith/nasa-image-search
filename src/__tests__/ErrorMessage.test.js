import React from 'react';
import { render } from '@testing-library/react';
import ErrorMessage from '../components/ErrorMessage';

const mockData = {
  errorStatus: 400,
  errorText: "Mock Text"
}

test('component ErrorMessage renders with error data passed in', () => {
  const { asFragment } = render(
    <ErrorMessage errorParams = {mockData} />
  );
  const component = asFragment();
  expect (component).toMatchSnapshot();
});

test('component ErrorMessage formats the returned string correctly', () => {
  const { getByText } = render(
    <ErrorMessage errorParams = {mockData} />
  );
  const message = getByText("400 Error: Mock Text");
  expect(message).toBeInTheDocument();
})