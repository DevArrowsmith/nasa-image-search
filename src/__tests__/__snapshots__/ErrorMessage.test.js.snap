// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component ErrorMessage renders with error data passed in 1`] = `
<DocumentFragment>
  <div
    data-testid="error-message"
  >
    <h2
      class="sc-AxjAm cmwgYd"
    >
      400 Error: Mock Text
    </h2>
  </div>
</DocumentFragment>
`;
